react/features/virtual-background/middleware.js

```diff
@@ -0,0 +1,26 @@
+// @flow
+
+import { MiddlewareRegistry } from '../base/redux';
+import { getLocalVideoTrack } from '../base/tracks';
+
+import { localTrackStopped } from './functions';
+
+/**
+ * Middleware which intercepts the desktop video type on
+ * virtual background. If the user stops the screen share
+ * then the default virtual background is set to 'none' option
+ *
+ * @param {Store} store - The redux store.
+ * @returns {Function}
+ */
+MiddlewareRegistry.register(store => next => action => {
+    const { dispatch, getState } = store;
+    const virtualSource = getState()['features/virtual-background'].virtualSource;
+    const currentLocalTrack = getLocalVideoTrack(getState()['features/base/tracks']);
+
+    if (virtualSource?.videoType === 'desktop') {
+        localTrackStopped(dispatch, virtualSource, currentLocalTrack.jitsiTrack);
```

Comment:
`currentLocalTrack` can be `undefined`  here if the user joins the call video muted.

Reply:
Added check for currentLocalTrack.

Reply:
With this change, the listener for `JitsiTrackEvents.LOCAL_TRACK_STOPPED` never gets added to the desktop track if there was no local video track when the share starts. Instead you can remove the check and replace `currentLocalTrack.jitsiTrack` with `currentLocalTrack?.jitsiTrack`.

Reply:
Fixed